### PR TITLE
Add bitmask backend

### PR DESCRIPTION
**Pull request by yohanchatelain**
- Applies a bitmask on floating-point numbers on the p-t last bits where p is the original precision and t is the virtual precision.
- Replaces floating-points operations by the following operation:
  fl(a op b) -> bitmask(bitmask(a) op bitmask(b)) where op is {+,-,*,/}
  - Three modes available:
    * ZERO: Set last bits to 0.
    * ONE: Set last bits to 1.
    * RAND: Applies a XOR with a random vector of bits.
- Adds DAZ and FTZ options:
  - DAZ: denormals-are-zero sets the denormal inputs to 0
  - FTZ: flush-to-zero sets the denormal output to 0
- Adds test for bitmask backend
  * Checks rand mode only
  * Rename test_backends into test_mca_backends

Change the random generator in test_mca_backends to reduce probability of taking nans

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -343,6 +343,64 @@ operations. It is heavily based on mcalib MPFR backend.
...
+The option `--precision-binary64=PRECISION` controls the virtual
+precision used for the floating point operations in double precision
+(respectively for single precision with --precision-binary32) It
+accepts an integer value that represents the virtual precision at
+which MCA operations are performed. Its default value is 53 for
+binary64 and 24 for binary32. For the Bitmask backend, the virtual
+precision corresponds to the number of preserved bits in the mantissa.
+
+The option `--seed` fixes the random generator seed. It should not
+generally be used except if one to reproduce a particular Bitmask
```
Comment (pablooliveira) [inline README.md]: except ~~if one~~ to reproduce

File: README.md [hunk 2]
```diff
@@ -343,6 +343,64 @@ operations. It is heavily based on mcalib MPFR backend.
 
 MCA-MPFR backend accepts the same options than the MCA backend.
 
+### Bitmask Backend (libinterflop_bitmask.so)
+
+The Bitmask backend implements a fast first order model of noise. It
+relies on bitmask operations to achieve very low overhead
```
Comment (pablooliveira) [inline README.md]: to achieve ~~very~~ low overhead

File: README.md (hunk 1 above)
Comment (yohanchatelain) [reply README.md]: Done.

File: README.md (hunk 2 above)
Comment (yohanchatelain) [reply README.md]: Done.

**Comment by pablooliveira** [comment]
Thanks :tada:

**Pull request merged**